From eb732b944eb38322795a984647975522d1ab2f70 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@st.com>
Date: Thu, 5 Sep 2019 16:32:36 +0200
Subject: [PATCH] board: stm32mp1: fixup the USB product id for fastboot

Select a specific product id used during USB enumeration
when the fastboot gadget is used.

For ST Microelectronics board we are using
the VID:PID = 0x0483:0x0AFB for fastboot
as defined in http://www.linux-usb.org/usb.ids

This patch avoid conflict when the same USB VID/PID is used for
fastboot and DFU (two different protocols associated to the same PID).

Signed-off-by: Patrick Delaunay <patrick.delaunay@st.com>
Change-Id: Id25154105bef11c7ed8031aed96e7c67b1805885
---
 board/st/stm32mp1/stm32mp1.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/board/st/stm32mp1/stm32mp1.c b/board/st/stm32mp1/stm32mp1.c
index f852b1e..0d04d03 100644
--- a/board/st/stm32mp1/stm32mp1.c
+++ b/board/st/stm32mp1/stm32mp1.c
@@ -388,10 +388,15 @@ int g_dnl_board_usb_cable_connected(void)
 }
 
 #define STM32MP1_G_DNL_DFU_PRODUCT_NUM 0xdf11
+#define STM32MP1_G_DNL_FASTBOOT_PRODUCT_NUM 0x0afb
+
 int g_dnl_bind_fixup(struct usb_device_descriptor *dev, const char *name)
 {
 	if (!strcmp(name, "usb_dnl_dfu"))
 		put_unaligned(STM32MP1_G_DNL_DFU_PRODUCT_NUM, &dev->idProduct);
+	else if (!strcmp(name, "usb_dnl_fastboot"))
+		put_unaligned(STM32MP1_G_DNL_FASTBOOT_PRODUCT_NUM,
+			      &dev->idProduct);
 	else
 		put_unaligned(CONFIG_USB_GADGET_PRODUCT_NUM, &dev->idProduct);
 
-- 
2.7.4

